apis/condition_set.go

```diff
@@ -207,6 +210,78 @@ func (r conditionsImpl) severity(t ConditionType) ConditionSeverity {
 	return ConditionSeverityInfo
 }
 
+// RemoveCondition removes the condition that matches the ConditionType
+func (r conditionsImpl) RemoveCondition(t ConditionType) {
```

Comment:
If you are adding `RemoveCondition` I would expect you would also add `AddCondition` to get back to the orignal setup.

Reply:
Won't AddCondition be a confusing one? All `Mark` functions add condition if they are not added earlier. RemoveCondition was added as this functionality was missing earlier.